### PR TITLE
`zstandard` import bandaid

### PR DESCRIPTION
**Pull request by mattwthompson**
The plan agreed to on 9/10 is

- [x] Import `zstandard` at the top of `openeye_wrapper.py`
- [x] Make a custom off-label build without `zstandard` constraint
  - [x] Look for `noarch/openff-toolkit-0.16.4-pyhc2fc6b4_3.conda`
- [x] Test locally with custom build (see below)

### CODE REVIEW

**Comment by mattwthompson** [comment]
This seems to work fine if snuck into a conda build as a standalone patch:


```shell
$ micromamba create --name with-1940 "openff-toolkit=0.16.4=pyhc2fc6b4_3" openeye-toolkits qcfractal -c conda-forge/label/openff-toolkit_rc -c conda-forge -c openeye -yq
warning  libmamba 'root_prefix' set with default value: /Users/mattthompson/micromamba
warning  libmamba [tk-8.6.13-h5083fa2_1] The following files were already present in the environment:
    - include/X11/DECkeysym.h
    - include/X11/HPkeysym.h
    - include/X11/Sunkeysym.h
    - include/X11/X.h
    - include/X11/XF86keysym.h
    - include/X11/Xatom.h
    - include/X11/Xfuncproto.h
    - include/X11/ap_keysym.h
    - include/X11/keysym.h
    - include/X11/keysymdef.h
warning  libmamba [xorg-libx11-1.8.10-h2321a68_0] The following files were already present in the environment:
    - include/X11/Xlib.h
    - include/X11/Xutil.h
    - include/X11/cursorfont.h
[~]
$ micromamba activate with-1940                                                      9:40:07
warning  libmamba 'root_prefix' set with default value: /Users/mattthompson/micromamba
(with-1940) [~]
$ python -c "from openeye import *; import qcportal"                                 9:40:15
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/mattthompson/micromamba/envs/with-1940/lib/python3.12/site-packages/qcportal/__init__.py", line 10, in <module>
    from .client import PortalClient
  File "/Users/mattthompson/micromamba/envs/with-1940/lib/python3.12/site-packages/qcportal/client.py", line 10, in <module>
    from qcportal.cache import DatasetCache, read_dataset_metadata
  File "/Users/mattthompson/micromamba/envs/with-1940/lib/python3.12/site-packages/qcportal/cache.py", line 22, in <module>
    import zstandard
  File "/Users/mattthompson/micromamba/envs/with-1940/lib/python3.12/site-packages/zstandard/__init__.py", line 39, in <module>
    from .backend_c import *  # type: ignore
    ^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: zstd C API versions mismatch; Python bindings were not compiled/linked against expected zstd version (10502 returned by the lib, 10506 hardcoded in zstd headers, 10506 hardcoded in the cext)
(with-1940) [~]
$ pytest --pyargs "openff.toolkit" -rf                                               9:40:33
==================================== test session starts =====================================
platform darwin -- Python 3.12.5, pytest-8.3.3, pluggy-1.5.0
rootdir: /Users/mattthompson
plugins: anyio-4.6.0, typeguard-4.3.0
collected 24383 items
```
test currently collected and running

File: openff/toolkit/utils/openeye_wrapper.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 __all__ = ("OpenEyeToolkitWrapper",)
 
 
+import zstandard  # noqa
```
Comment (j-wags) [inline openff/toolkit/utils/openeye_wrapper.py:9]: zstandard isn't a hard dep of our package - I believe it's pulled in by qcportal. So this import should be wrapped in a `try/except`

```suggestion
try:
    import zstandard  # noqa
except ImportError:
    pass
```

**Comment by mattwthompson** [comment]
+ local tests with and without QCPortal installed

File: openff/toolkit/utils/openeye_wrapper.py [hunk 2]
```diff
@@ -6,6 +6,10 @@
 __all__ = ("OpenEyeToolkitWrapper",)
 
 
+try:
```
Comment (j-wags) [inline openff/toolkit/utils/openeye_wrapper.py:9]:
```suggestion
# See https://github.com/conda-forge/openff-toolkit-feedstock/issues/86
try:
```

**Comment by mattwthompson** [comment]
I'm going to defer a large collection of RDKit-related updates (https://github.com/openforcefield/openff-toolkit/pull/1942) to a subsequent release. Every issue I've run into has either been something that is upstream's (and not our) responsibility to address or something that requires changes in test/automation code, not source code.

**Pull request merged**